R/cat_add_data.R

```diff
@@ -7,15 +7,113 @@
 #' sources.
 #'
 #' @param x an object of class [elic_cat].
-#' @param mechanism
+#' @param data_source either a [`data.frame`][base::data.frame] or
+#' [`tibble`][tibble::tibble], a string with the path to a _csv_ or _xlsx_ file,
+#' or anything accepted by the [read_sheet()][googlesheets4::read_sheet]
+#' function.
+#' @param mechanism integer indicating what mechanism the data belongs to.
 #' @inheritParams elic_cont_add_data
 #'
+#' @section Data Format:
+#'
+#' For each mechanism, data are expected to have five columns, built as follows:
+#' * The first column of the data should hold the names of the experts. The name
+#' of each expert should be repeated as many times as there are impact levels
+#' and sites considered in the elicitation
+#' (i.e. each expert should appear \eqn{number\ of\ levels
+#' \cdot number\ of\ sites} times).
+#' * The second column should be the names of the levels considered in the
+#' eliciation. Each level should be repeated as many times as there are sites
+#' considered.
+#' * The third column should hold the names of the sites considered in the
+#' study. The name of each site should be repeated as many times as the number
+#' of levels.
+#' * The fourth column should be the experts confidence levels (given in
+#' percents). Experts should each give one confidence level per site, so each
+#' confidence level should be repeated for all levels at each site.
+#' * The final column should be the estimates of each expert.
+#'
+#' @section Data cleaning:
+#'
+#' When data are added to the [elic_cat] object, first names are standardised
+#' by converting capital letters to lower case, and by removing any whitespaces
+#' and punctuation. Then, data are anonymised by converting names to short sha1
+#' hashes. In this way, sensible information collected during the elicitation
+#' process never reaches the [elic_cat] object.
+#'
+#' If the data are imported from _Google Sheets_, `elic_cat_add_data()`
+#' performs additional data cleaning operations. This is relevant when data are
+#' collected with Google Forms because, for example, there could be multiple
+#' submission by the same expert or a different decimal separator could be used.
+#' When data are collected with Google Form, a column with the date and time is
+#' recorded. First, the function checks for multiple submissions and if present,
+#' only the last submission is retained. Second, the function removes the column
+#' with the timestamp. Then it checks for consistency of the decimal separator,
+#' i.e. commas _,_ are replaced with periods _._. Finally, columns four and five
+#' (which contain the confidence levels and estimates) are forced to numeric.
```

Comment:
Lines 43-53: I would remove this part. I don't think you can collect these data with google form. I mean, you can but after you need to format them to be imported in R. My idea is to not implement this data cleaning for the categorical data, at least not at this stage since we don't have enough time.

Reply:
ok 👍